### ISSUE TITLE
Android: android-extension for application & library don't work

### ISSUE THREAD

**Issue by StefMa** [gradle/kotlin-dsl#844]
I have an Android project where I apply the `com.android.application` plugin via the `plugins{}` block:
```
plugins {
    id("com.android.application")
}
```
For that I've modified the settings.gradle:
```
pluginManagement {
    repositories {
        gradlePluginPortal()
        jcenter()
        google()
    }
    resolutionStrategy {
        eachPlugin {
            if (requested.id.id.startsWith("com.android.")) {
                useModule("com.android.tools.build:gradle:3.1.2")
            }
        }
    }
}
```

Everything works so far. 
But when I create a new module and apply the `com.android.library` plugin I get:
```
com.android.build.gradle.LibraryExtension_Decorated cannot be cast to com.android.build.gradle.AppExtension
Possible causes for this unexpected error include:<ul><li>Gradle's dependency cache may be corrupt (this sometimes occurs after a network connection timeout.)
Re-download dependencies and sync project (requires network)</li><li>The state of a Gradle build process (daemon) may be corrupt. Stopping all Gradle daemons may solve this problem.
Stop Gradle build processes (requires restart)</li><li>Your project may be using a third-party plugin which is not compatible with the other plugins in the project or the version of Gradle requested by the project.</li></ul>In the case of corrupt Gradle processes, you can also try closing the IDE and then killing all Java processes.
```

I do that in the library build.gradle.kts:
```
plugins {
    id("com.android.library")
}
```

### Expected Behavior
I want to apply both plugins in the same manner like I do before (in the groovy version).

### Current Behavior
I can't sync Gradle...

### Steps to Reproduce (for bugs)
Just create a default andorid project. Use the settings.gradle from above and move to build.gradle.kts files.
Then create a new `com.android.library` module...

### Your Environment
Gradle 4.7

**Comment by gildor** [comment]
Are you sure that you do not apply both plugins in the same project? Because my project works without problem with this config, maybe some regression.

**Comment by StefMa** [comment]
Tested it with this sample project here and it worked.
See: https://github.com/gradle/kotlin-dsl/compare/d13ff734053a68e8c5991cf97d535abad60b5ecd...StefMa:android-multiModule?expand=1

Strange. A project at work haven't worked 🤔 
Maybe I had some caching issues. Closed since it should normally work 😂

**Comment by StefMa** [comment]
Sry, I have to reopen the issue because I have still that issue.
What do you mean exactly with "the same model" @gildor ?

This is a test project:
[Issue.zip](https://github.com/gradle/kotlin-dsl/files/1987859/Issue.zip)
It contain a android project along with a Dockerfile.
To test this issue just do the following:
1. Extract it
2. `cd` to the dir
```
docker build . -t TestImage
docker run TestImage
```

Then see that everything runs and fail with:
```
> com.android.build.gradle.LibraryExtension_Decorated cannot be cast to com.android.build.gradle.AppExtension
```

**Comment by eskatos** [comment]
Thanks for the reproducer @StefMa, I can reproduce locally.
It definitely looks like a caching issue and may be related to #854

**Comment by bamboo** [comment]
Fixed in eb05db120de014cd91cec33dff04f113ac90ea3f.
